dashboard/app/jobs/evaluate_rubric_job.rb

```diff
@@ -390,11 +399,21 @@ def validate_ai_config_for_lesson(lesson_s3_name, code)
     end
   end
 
-  private def merge_confidence_levels(ai_evaluations, ai_confidence_levels)
+  private def merge_confidence_levels(ai_evaluations, ai_confidence_levels_pass_fail, ai_confidence_levels_exact_match)
     ai_evaluations.map do |evaluation|
       learning_goal = evaluation['Key Concept']
-      confidence_level = ai_confidence_levels[learning_goal]
-      evaluation.merge('Confidence' => confidence_s_to_i(confidence_level))
+      confidence_pass_fail = ai_confidence_levels_pass_fail[learning_goal]
+      evaluation['Confidence-Pass-Fail'] = confidence_s_to_i(confidence_pass_fail)
+
+      if ai_confidence_levels_exact_match
+        label = evaluation['Label']
+        confidence_exact_match = ai_confidence_levels_exact_match[learning_goal][label]
+        unless confidence_exact_match
+          raise "No confidence_exact_match for learning goal: #{learning_goal}, label: #{label} evaluation: #{JSON.pretty_generate(evaluation)} ai_confidence_levels_exact_match: #{JSON.pretty_generate(ai_confidence_levels_exact_match)}"
+        end
+        evaluation['Confidence-Exact-Match'] = confidence_s_to_i(confidence_exact_match)
+      end
+      evaluation
```

Comment:
For my own education- what's the purpose of this evaluation statement?

Reply:
this is an implicit return value. https://franzejr.github.io/best-ruby/idiomatic_ruby/implicit_return.html in this particular case, because we are doing a map operation, the original object needs to be returned in order for the result to reflect the additions to the hashmap. then the entire hashmap is implicitly returned as the return value for the enclosing method.